specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/storageTaskAssignments.json

```diff
@@ -0,0 +1,842 @@
+          "items": {
+            "type": "string"
+          },
+          "description": "List of target prefixes to exclude"
+        },
+        "reservedPathPrefix": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "List of reserved path prefixes"
+        }
+      },
+      "required": [
+        "prefix"
+      ],
+      "description": "Target for storage task assignment execution"
+    },
+    "ExecutionTrigger": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "RunOnce",
+            "OnSchedule"
+          ],
+          "x-ms-enum": {
+            "name": "TriggerType",
+            "modelAsString": false
+          },
+          "description": "The trigger type of the storage task assignment execution"
+        },
+        "parameters": {
+          "$ref": "#/definitions/TriggerParameters",
+          "description": "The trigger parameters of the storage task assignment execution"
+        }
+      },
+      "required": [
+        "type",
+        "parameters"
+      ],
+      "description": "Execution trigger for storage task assignment"
+    },
+    "TriggerParameters": {
+      "type": "object",
+      "properties": {
+        "startFrom": {
+          "type": "string",
+          "description": "When to start task execution. This is a required field when ExecutionTrigger.properties.type is 'OnSchedule'; this property should not be present when ExecutionTrigger.properties.type is 'RunOnce'"
+        },
+        "interval": {
+          "type": "integer",
+          "minimum": 1,
+          "format": "int32",
+          "description": "Run interval of task execution. This is a required field when ExecutionTrigger.properties.type is 'OnSchedule'; this property should not be present when ExecutionTrigger.properties.type is 'RunOnce'"
+        },
+        "intervalUnit": {
+          "type": "string",
+          "enum": [
+            "days"
+          ],
+          "x-ms-enum": {
+            "name": "IntervalUnit",
+            "modelAsString": false
+          },
+          "description": "Run interval unit of task execution. This is a required field when ExecutionTrigger.properties.type is 'OnSchedule'; this property should not be present when ExecutionTrigger.properties.type is 'RunOnce'"
+        },
+        "endBy": {
+          "type": "string",
+          "description": "When to end task execution. This is a required field when ExecutionTrigger.properties.type is 'OnSchedule'; this property should not be present when ExecutionTrigger.properties.type is 'RunOnce'"
+        },
+        "startOn": {
+          "type": "string",
+          "description": "When to start task execution. This is an optional field when ExecutionTrigger.properties.type is 'RunOnce'; this property should not be present when ExecutionTrigger.properties.type is 'OnSchedule'"
+        }
```

Comment:
why do you have both 'startFrom' and 'startOn'??

Reply:
runOnce - allows you to chose when to startOn ; StartFrom and EndBy does not apply here

onSchedule - Allows you to provide an interval, StartFrom and EndBy ; StartOn Does not apply here

tricia - Kindly check if we can make the description more descriptive

Reply:
I got that you intend them to be used differently, but to me the distinction between 'StartFrom' and 'StartOn' seems very arbitrary. Aren't they basically the same thing, which is a date/time when you want something to start happening? And therefore very easy for anyone to confuse? Imagine I'm typing in my IDE and I see there are two possible completions, StartFrom and StartOn. My immediate reaction is 'why?'

Reply:
In the description I mention:
"startFrom": When to start task execution. This is a **required field when ExecutionTrigger.properties.type is 'OnSchedule'**; this property **should not be present when ExecutionTrigger.properties.type is 'RunOnce'**
vs
"startOn": When to start task execution. This is an **optional field when ExecutionTrigger.properties.type is 'RunOnce'**; this property **should not be present when ExecutionTrigger.properties.type is 'OnSchedule'**

These properties are based on the triggerType. You cannot have "startOn" present when trigger type is 'OnSchedule', this will lead to a server error.

Reply:
>>I got that you intend them to be used differently, but to me the distinction between 'StartFrom' and 'StartOn' seems very arbitrary. Aren't they basically the same thing, which is a date/time when you want something to start happening? And therefore very easy for anyone to confuse? Imagine I'm typing in my IDE and I see there are two possible completions, StartFrom and StartOn. My immediate reaction is 'why?'

So this turns out to be more intuitive for our customers, based on the user study performed by Storage Actions PMs ; (We can tag them if required)
Differentiating with action verbs, startOn and startFrom has been well accepted by the customers in preview..